resources/Microsoft.Windows.Setting.WindowsUpdate/Microsoft.Windows.Setting.WindowsUpdate.psm1

```diff
@@ -0,0 +1,377 @@
+
+.PARAMETER IsContinuousInnovationOptedIn
+    Indicates whether the device is opted in for continuous innovation updates. This is the setting in Windows Update settings -> Get the latest updates as soon as they're available.
+
+.PARAMETER AllowMUUpdateService
+    Indicates whether the Microsoft Update service is allowed. This is the setting in Windows Update settings -> Advanced options -> Receive updates for other Microsoft products.
+
+.PARAMETER IsExpedited
+    Indicates whether the updates are expedited. This is the setting in Windows Update settings -> Advanced options -> Get me up to date.
+
+.PARAMETER AllowAutoWindowsUpdateDownloadOverMeteredNetwork
+    Indicates whether automatic Windows Update downloads are allowed over metered networks. This is the setting in Windows Update settings -> Advanced options -> Download updates over metered connections.
+
+.PARAMETER RestartNotificationsAllowed
+    Indicates whether restart notifications are allowed. This is the setting in Windows Update settings -> Advanced options -> Notify me when a restart is required to finish updating.
+
+.PARAMETER SmartActiveHoursState
+    Indicates whether smart active hours are enabled.
+
+.PARAMETER UserChoiceActiveHoursEnd
+    The end time for user-chosen active hours.
+
+.PARAMETER UserChoiceActiveHoursStart
+    The start time for user-chosen active hours.
+
+.PARAMETER DownloadMode
+    The download mode for updates. Valid values are 0, 1, and 3. This is the setting in Windows Update settings -> Advanced options -> Delivery Optimization -> Allow downloads from other PCs.
+
+.PARAMETER DownloadRateBackgroundBps
+    The background download rate in bits per second.
+
+.PARAMETER DownloadRateForegroundBps
+    The foreground download rate in bits per second.
+
+.PARAMETER DownloadRateBackgroundPct
+    The background download rate as a percentage.
+
+.PARAMETER DownloadRateForegroundPct
+    The foreground download rate as a percentage.
+
+.PARAMETER UploadLimitGBMonth
+    The upload limit in gigabytes per month.
+
+.PARAMETER UpRatePctBandwidth
+    The upload rate as a percentage of bandwidth.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name WindowsUpdate -Method Get -ModuleName Microsoft.Windows.Setting.WindowsUpdate -Property @{}
+
+    This command gets the current Windows Update settings.
+#>
+[DSCResource()]
+class WindowsUpdate
+{
+    # Key required. Do not set.
+    [DscProperty(Key)] 
+    [string] $SID
+    
+    [DscProperty()]
+    [nullable[bool]] $IsContinuousInnovationOptedIn
+
+    [DscProperty()]
+    [nullable[bool]] $AllowMUUpdateService
+
+    [DscProperty()]
+    [nullable[bool]] $IsExpedited
+
+    [DscProperty()]
+    [nullable[bool]] $AllowAutoWindowsUpdateDownloadOverMeteredNetwork
+
+    [DscProperty()]
+    [nullable[bool]] $RestartNotificationsAllowed
+
+    [DscProperty()]
+    [nullable[bool]] $SmartActiveHoursState
+
+    [DscProperty()]
+    [nullable[int]] $UserChoiceActiveHoursEnd
```

Comment:
Inatead of having the assert function, just validate the parameter using `[ValidateRange(0,24)]` and maybe `[AllowNull()]`if its needed